pennylane/tape/qnode.py

```diff
@@ -249,21 +252,39 @@ def _validate_backprop_method(device, interface):
         # determine if the device supports backpropagation
         backprop_interface = device.capabilities().get("passthru_interface", None)
 
+        # determine if the device has any child devices that support backpropagation
+        backprop_devices = device.capabilities().get("passthru_devices", None)
+
         if getattr(device, "cache", 0):
             raise qml.QuantumFunctionError(
                 "Device caching is incompatible with the backprop diff_method"
             )
 
         if backprop_interface is not None:
+            # device supports backpropagation natively
 
             if interface == backprop_interface:
-                return JacobianTape, None, "backprop"
+                return JacobianTape, interface, "backprop", device
 
             raise qml.QuantumFunctionError(
                 f"Device {device.short_name} only supports diff_method='backprop' when using the "
                 f"{backprop_interface} interface."
             )
 
+        if getattr(device, "analytic", False) and backprop_devices is not None:
+            # device is analytic and has child devices that support backpropagation natively
+
+            if interface in backprop_devices:
+                # TODO: need a better way of passing existing device init options
+                # to a new device?

```

Comment:
Good point :thinking:

I guess one option could be to define a `to_tf` interface in `DefaultQubit` that converts it to its TF version? This would also solve the problem of the user not being able to access the newly defined device.

Reply:
That's a good idea... but I'm not sure how to implement it 🤔 

Changing the device requires changing the device class. We could do
```python
self.__class__ = passthru_device
```

but this is probably quite unsafe, especially if the `__init__` methods differ between the two.

Reply:
Right! So is this ok to do for tapes (e.g., in `TFInterface.apply()`) because the `__init__` methods are the same?

Reply:
yeah, `TFInterface`, `AutogradInterface` were designed from scratch with this capability in mind (they are mixin classes). Whereas we never designed the devices to be swappable 🤔

Reply:
Some details: https://stackoverflow.com/a/13280789/10958457